### ISSUE TITLE
Adding ability to use IAM roles 

### ISSUE THREAD

**Issue by GnatorX** [CheckPointSW/Cuckoo-AWS#28]
##### This can be reproduced by:
NA

##### The log, error, files etc can be found at:
NA

https://github.com/CheckPointSW/Cuckoo-AWS/pull/26

**Comment by chkp-mostafaw** [comment]
Hi @GnatorX 
Do you mean assigning IAM role to the EC2?

**Comment by GnatorX** [comment]
Yup

**Comment by GnatorX** [comment]
Not exactly. This is for if you already gave your EC2 Instance an IAM instance profile with an IAM role, this PR will pick up that IAM role

**Comment by GnatorX** [comment]
leverage the existing cred retrieval behavior https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html#credentials-chain
